src/util/config.rs

```diff
@@ -113,8 +117,116 @@ pub fn parse_readable_int(size: &str) -> Result<i64, ParseConfigError> {
         "m" => Ok((num * (MINTUE as f64)) as i64),
         "h" => Ok((num * (HOUR as f64)) as i64),
 
-        _ => Err(ParseConfigError::ReadableNumber),
+        _ => Err(ConfigError::ReadableNumber),
+    }
+}
+
+#[cfg(unix)]
+pub fn check_max_open_fds(expect: u64) -> Result<(), ConfigError> {
+    use libc;
+
+    let fd_limit = unsafe {
+        let mut fd_limit = ::std::mem::zeroed();
+        if 0 != libc::getrlimit(libc::RLIMIT_NOFILE, &mut fd_limit) {
+            return Err(ConfigError::Limits("check_max_open_fds failed".to_owned()));
+        }
+
+        fd_limit
+    };
+
+    if fd_limit.rlim_cur < expect {
+        return Err(ConfigError::Limits(format!("the maximum number of open file descriptors is \
+                                                too small, got {}, expect greater or equal to \
+                                                {}",
+                                               fd_limit.rlim_cur,
+                                               expect)));
     }
+
+    Ok(())
+}
+
+#[cfg(not(unix))]
+pub fn check_max_open_fds(expect: u64) -> Result<(), ConfigError> {
+    Ok(())
+}
+
+/// `check_kernel_params` checks kernel parameters, following are checked so far:
+///   - `net.core.somaxconn` should be greater or equak to 32768.
+///   - `net.ipv4.tcp_syncookies` should be 0
+///   - `vm.swappiness` shoud be 0
+///
+/// Note that: It only works **Linux** platform.
+#[cfg(target_os = "linux")]
+pub fn check_kernel_params() -> Result<(), ConfigError> {
```

Comment:
It adds two new functions,

 1. `check_max_open_fds`, works on both Linux and macOS
 2. `check_kernel_params`, Linux only.

So, I think `mod unix` does not fit.

I dont use another file because I dont want `quick_error` again.